src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -1112,6 +1145,31 @@ private IEnumerable<ConfigurationSetting> GetCurrentKeyValueCollection(string ke
             return currentKeyValues;
         }
 
+        private void UpdateClientBackoffStatus(Uri endpoint, bool successful)
+        {
+            if (!_configClientBackoffs.TryGetValue(endpoint, out ConfigurationClientBackoffStatus clientBackoffStatus))
+            {
+                clientBackoffStatus = new ConfigurationClientBackoffStatus();
+            }
+
+            if (successful)
+            {
+                clientBackoffStatus.BackoffEndTime = DateTimeOffset.UtcNow;
+
+                clientBackoffStatus.FailedAttempts = 0;
+            }
+            else
+            {
+                clientBackoffStatus.FailedAttempts++;
+
+                TimeSpan backoffDuration = _options.MinBackoffDuration.CalculateBackoffDuration(FailOverConstants.MaxBackoffDuration, clientBackoffStatus.FailedAttempts);
```

Comment:
https://github.com/Azure/AppConfiguration-DotnetProvider/blob/e9a3aa5b49958bd6190fb91c7a16d10ac46e3967/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs#L73

Here basically the backoff time = minDuration * (1<<attemps) when attempts >=2.
attemps: 1, 2, 3, 4...
backoffTime( x minDuration): 1x, **4x**, 8x, 16x...

Should we change it to `1<<(attemps -1)`, so the backoff time becomes 1x,2x,4x,8x, ...?

Reply:
That is a good idea. I think this happens because we optimized to return when attempts=1, so we never get the 2x backoff duration.

Reply:
Makes sense, just gonna subtract one from attempts in that code then.